extensions/stackdriver/log/logger.cc

```diff
@@ -120,70 +105,149 @@ Logger::Logger(const ::Wasm::Common::FlatNode& local_node_info,
           flatbuffers::GetString(rev_iter->value());
     }
   }
+}
+
+// Helper methods to fill source Labels.
+void fillSourceLabels(
+    const ::Wasm::Common::FlatNode& source_node_info,
+    google::protobuf::Map<std::string, std::string>* label_map) {
+  (*label_map)["source_name"] = flatbuffers::GetString(source_node_info.name());
+  (*label_map)["source_workload"] =
+      flatbuffers::GetString(source_node_info.workload_name());
+  (*label_map)["source_namespace"] =
+      flatbuffers::GetString(source_node_info.namespace_());
+  // Add destination app and version label if exist.
+  const auto local_labels = source_node_info.labels();
+  if (local_labels) {
+    auto version_iter = local_labels->LookupByKey("version");
+    if (version_iter) {
+      (*label_map)["source_version"] =
+          flatbuffers::GetString(version_iter->value());
+    }
+    // App label is used to correlate workload and its logs in UI.
+    auto app_iter = local_labels->LookupByKey("app");
+    if (app_iter) {
+      (*label_map)["source_app"] = flatbuffers::GetString(app_iter->value());
+    }
+    if (label_map->find("source_canonical_service") == label_map->end()) {
+      setSourceCanonicalService(source_node_info, label_map);
+    }
+    auto rev_iter = local_labels->LookupByKey(
+        Wasm::Common::kCanonicalServiceRevisionLabelName.data());
+    if (rev_iter) {
+      (*label_map)["source_canonical_revision"] =
+          flatbuffers::GetString(rev_iter->value());
+    }
+  }
+}
+
+}  // namespace
+
+using google::protobuf::util::TimeUtil;
+
+// Name of the server access log.
+constexpr char kServerAccessLogName[] = "server-accesslog-stackdriver";
+// Name of the client access log.
+constexpr char kClientAccessLogName[] = "client-accesslog-stackdriver";
+
+void Logger::initializeLogEntryRequest(
+    const flatbuffers::Vector<flatbuffers::Offset<Wasm::Common::KeyVal>>*
+        platform_metadata,
+    const ::Wasm::Common::FlatNode& local_node_info, bool outbound) {
+  auto log_entry_type = GetLogEntryType(outbound);
+  log_entries_request_map_[log_entry_type]->request =
+      std::make_unique<google::logging::v2::WriteLogEntriesRequest>();
+  log_entries_request_map_[log_entry_type]->size = 0;
+  auto log_entries_request =
+      log_entries_request_map_[log_entry_type]->request.get();
+  std::string log_name = outbound ? kClientAccessLogName : kServerAccessLogName;
+  log_entries_request->set_log_name("projects/" + project_id_ + "/logs/" +
+                                    log_name);
+
+  std::string resource_type = outbound ? Common::kPodMonitoredResource
```

Comment:
nit: const std::string&

Reply:
Thanks fixed!

Reply:
Cant change to const ref as we assigning resource_type based on cluster name key below...